Tests/FunctionalTests/DependencyResolutionTests.swift

```diff
@@ -65,4 +65,62 @@ class DependencyResolutionTests: XCTestCase {
             XCTAssertEqual(output, "♣︎K\n♣︎Q\n♣︎J\n♣︎10\n♣︎9\n♣︎8\n♣︎7\n♣︎6\n♣︎5\n♣︎4\n")
         }
     }
+
+    func testMirrors() {
```

Comment:
😓

Reply:
Hmm, was there no test at all before?!

Reply:
there was a test that uses all mocks, no end-to-end test